src/bigunit.ts

```diff
@@ -286,6 +289,12 @@ export class BigUnit {
    * @returns BigUnit with the maximum value
    */
   public static max(unit1: BigUnit, unit2: BigUnit): BigUnit {
+    if (unit1.eq(unit2)) {
+      if (unit1.precision === unit2.precision) {
+        return unit1.name.localeCompare(unit2.name) <= 0 ? unit1 : unit2;
+      }
+      return unit1.precision > unit2.precision ? unit1 : unit2;
+    }
```

Comment:
Sorry Taylor but I just don't think this is useful functionality, and will have an impact on performance. Let's keep this library lean.
```suggestion
```

Reply:
Oh you mean the name part? Sorry I thought you meant the shared function

Reply:
Fair enough, I am finding conflicts between the ticket and your intended functionality. This ticket is a bit different than others as you are the product owner for BigUnit, but suggestions to change the code are being made without your consultation. I'll pause on this guy and we can chat about it when you are in next :)